### PR TITLE
Fixes #15167: Invalid detection of empty strings for edition in Techniques

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15167

### CODE REVIEW

**Comment by ncharles** [comment]
I haven't tested it yet

File: techniques/fileDistribution/checkGenericFileContent/8.0/checkGenericFileContent.st [hunk 1]
```diff
@@ -120,7 +120,7 @@ bundle agent check_generic_file_content_RudderUniqueID {
 
       # Check that value has been set before to edit the file
       "generic_file_content_modification_destination_defined_${index}" not => strcmp("${generic_file_content_modification_destination[${index}]}", "");
-      "edit_content_${index}" not => strcmp("${generic_file_content_payload[${index}]}", "");
```
Comment (ncharles) [inline techniques/fileDistribution/checkGenericFileContent/8.0/checkGenericFileContent.st]: we should keep both, and report if both classes are unset

**Comment by ncharles** [comment]
Commit modified

File: techniques/fileDistribution/checkGenericFileContent/8.0/checkGenericFileContent.st [hunk 2]
```diff
@@ -242,7 +249,7 @@ bundle agent check_generic_file_content_RudderUniqueID {
 
       # File edition - enforce only at creation
       "any" usebundle => rudder_common_report_index("checkGenericFileContent", "result_success", "${generic_file_content_uuid[${index}]}", "File", "${generic_file_content_path[${index}]}", "The file ${generic_file_content_path[${index}]} already exists, skipping content editing", "${index}"),
-        ifvarclass => "edit_content_${index}.creation_only_but_already_exists_${index}";
+        ifvarclass => "no_edit_content_${index}.creation_only_but_already_exists_${index}";
```
Comment (amousset) [inline techniques/fileDistribution/checkGenericFileContent/8.0/checkGenericFileContent.st]: reversed logic?

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**